[WIP] Added: Command palette component

rishimohan opened the pull request:
## What kind of change does this PR introduce?

Fixes https://github.com/supabase/supabase/issues/6986, by adding a command palette or Cmd + K search component.

## What is the new behavior?

Command Palette component can be quickly used to search through the docs, or pages from studio.

## Additional context

The PR is currently WIP. Checklist:

- [ ]  Add storybook prop documentation for the Command Palette
- [ ]  Update search results in palette to highlight matching terms

Screenshots:

<img width="716" alt="image" src="https://user-images.githubusercontent.com/14031295/171315774-8ed6750d-4867-4205-821f-b061313a67e8.png">

<img width="699" alt="image" src="https://user-images.githubusercontent.com/14031295/171315757-b671f9b5-9ebc-4ea6-b133-39ff61d92ca4.png">

MildTomato commented on src/components/CommandPalette/CommandPalette.js:
```diff
@@ -0,0 +1,113 @@
+            onKeyDown={(e) => handleShortcuts(e)}
+            value={term}
+            onChange={onSearch}
+            className="text-sm md:text-base pr-3 py-3 w-full focus:outline-none bg-transparent placeholder-dark-300 dark:placeholder-dark-400 dark:text-white"
```
@rishimohan You're probably going to have to move all the CSS classes into the theme file.

MildTomato replied:
using this defaultTheme

https://github.com/supabase/ui/blob/alpha/src/lib/theme/defaultTheme.ts

MildTomato replied:
then import classes like this
https://github.com/supabase/ui/blob/87776677af000bed91998403b8d3f936b8b10592/src/components/Button/Button.tsx#L88

MildTomato commented on src/components/CommandPalette/CommandPalette.js:
```diff
@@ -0,0 +1,107 @@
+            style={{ margin: 0 }}
+          />
+          {results?.list?.length > 0 ? (
+            <div className="text-gray-800 px-4 text-sm whitespace-nowrap dark:text-dark-400">
```
everywhere you've used 'gray' you'll need to replace with 'scale' as well, so when the theme colors swap, the new colors are used.

rishimohan commented:
@MildTomato Thanks for the feedback. I've moved the styles to CSS file, also updated colors to use `scale` values. Understanding new theme and colors was slightly confusing for me so there's possibility I might have missed something 😅

MildTomato commented:
### Supabase UI is being deprecated.

As part of that, we are closing Pull Requests in this repo for now.

If you'd like to still make these changes, you could consider making these changes in https://github.com/supabase/supabase/tree/master/packages/ui

---

We are moving the components to the main mono repo at [https://github.com/supabase/supabase](https://github.com/supabase/supabase).

The auth component has been moved to it's own repo and package. [https://github.com/supabase-community/auth-ui](https://github.com/supabase-community/auth-ui). You can also read the [docs for auth ui here](https://supabase.com/docs/guides/auth/auth-helpers/auth-ui).
